primer PR!!

franciscoMigliardi24 opened the pull request:
se leve con mi ser, estoy viviendo a galletitas de arroz y cafe
pd: casi feliz san valentin

franciscoMigliardi24 committed:
read me compleatado

franciscoMigliardi24 committed:
script para crear un usuario nuevo

franciscoMigliardi24 committed:
Merge pull request #1 from franciscoMigliardi24/crear-usuario

script para crear un usuario nuevo

franciscoMigliardi24 committed:
creacion de cliente

franciscoMigliardi24 committed:
Merge pull request #2 from franciscoMigliardi24/crear-cliente

creacion de cliente

franciscoMigliardi24 committed:
editar un cliente

franciscoMigliardi24 committed:
Merge pull request #3 from franciscoMigliardi24/editar-cliente

editar un cliente

franciscoMigliardi24 committed:
arreglos de creacion
se buscaba un valor previo para despues ingresar el valor correspondiente

didierrenardub commented on README.md:
```diff
@@ -0,0 +1,3 @@
+correr en la consola 
+"pip install selenium"
+apretar F5 para ejecutar cada test
```
Estas instrucciones están incorrectas; al presionar F5 en la consola, nada sucede.

franciscoMigliardi24 replied:
perdon, es F5 en visual

didierrenardub replied:
Qué Visual? Visual Studio? Si es Visual Studio, qué variante? C#? C++? F#? JSP? ASP? Adicionalmente, qué versión? 2013? 2015? 2017? 2019?
Alternativamente, también podría ser Visual Code...

franciscoMigliardi24 replied:
touché, you win this time

didierrenardub commented on crearCliente.py:
```diff
@@ -0,0 +1,79 @@
+
+
+if "login.html" in driver.title:
+  print("Funciona")
```
Estos mensajes se los había puesto a Rocío mientras le explicaba cómo usar Selenium; sólo debería informarse en el caso de las pruebas que encuentran errores y tienen que ser mensajes más descriptivos: en dónde se encontró el error, de qué se trata el error, etc..

Este comentario aplica para todos los casos en que sucede

franciscoMigliardi24 replied:
> Estos mensajes se los había puesto a Rocío 

es correcto, le pedi a rocio lo que tenia para poder proseguir con lo que estaba haciendo

voy a corregeir esto

didierrenardub commented on crearCliente.py:
```diff
@@ -0,0 +1,79 @@
+#enviar cliente creado
+driver.find_element_by_class_name('boton-enviar').click()
+
+time.sleep(2)
```
Este sleep está de más. Aplicar comentario a todos los lugares donde sucede esto (en los otros scripts)

didierrenardub commented on crearCliente.py:
```diff
@@ -0,0 +1,79 @@
+
+#ingresar palabra Proofreading
+element = driver.find_element_by_name("palabraProofreading")
+element.send_keys("1") 
```
Para evitar la constante repetición de `driver.find_element_by_name(elemento)` y `element.send_keys(teclas)`, deberías declarar una función que encapsule lo genérico para luego pasarle los argumentos específicos necesarios y así quede más sintético cada input.

didierrenardub commented on crearCliente.py:
```diff
@@ -0,0 +1,79 @@
+
+
+#enviar cliente creado
+driver.find_element_by_class_name('boton-enviar').click()
```
Estás creando el cliente pero no estás verificando de ninguna manera que haya sido correctamente creado.

didierrenardub commented on crearCliente.py:
```diff
@@ -0,0 +1,79 @@
+#enviar cliente creado
+driver.find_element_by_class_name('boton-enviar').click()
+
+time.sleep(2)
```
Faltan, además, casos de prueba para comprobar el funcionamiento del manejo de errores de la validación de los campos. Sucede lo mismo en otros scripts.

franciscoMigliardi24 replied:
no estaria entendiendo, como es esto?

didierrenardub replied:
Según lo que vi del código, no debería dejarte avanzar con la creación de un cliente o lo que sea en el caso de que falten completar campos, por ejemplo...

franciscoMigliardi24 replied:
ahora entiendo lo que decis, se me ocurren un par de ideas para hacerlo pero tengo que ver si son viables

franciscoMigliardi24 replied:
>en el caso de que falten completar campos

para eso estan las validaciones de front end ;) la cuales no permiten que se haga click.
ahora lo que se me habia ocurrido, pero no estaria encontrando la forma es de contar la cantidad de imputs y a partir de ahi ver si se agrego uno nuevo.
pero tenemos 2 problemas
1- de esta forma si se modifica un campo (de nombre a nombre de la abuela, por ejemplo) siguen estando la misma cantidad de imputs. - pero lo bueno es que busca por id o por nombre el .py y no podria identificar el elemento el cual fue modificado.
2- si faltara completar un campo este nunca seria completado y hay que modificar el .py - peero no se podria enviar el cliente/usuario ya que no apareceria el boton, gracias a las validaciones front end.

lo que se me ocurrio para esto es hacer un "try except" (try catch) al momento de ingresar los valores, mostrando un mensaje de error cuando no se encuentre el elemento o cuando falle algo

didierrenardub replied:
Pero alguien tiene que verificar que el comportamiento sea el esperado, por lo tanto, si bien las validaciones están en el front end, los tests automáticos con Selenium tienen que comprobar que esas validaciones estén funcionando correctamente.

Es que si cambia la especificación, es natural que también deban ser actualizadas las pruebas.

franciscoMigliardi24 replied:
>Pero alguien tiene que verificar que el comportamiento sea el esperado

espero que sea el que ejecuta el test jajajaj

la idea era hacer test mas atomicos como los que estan aca en el pr, uno para crear usuario, uno para crear cliente, uno para modificar etc etc...
entonces un test que se ejecuta solo verifica lo que el nombre del test diga "crearCliente.py" solo crea un cliente, y asi..

por eso la idea del "try except" (try catch, no me gusta el nombre de py) que muestre en consola (donde se ejecuta el test) que algo salio bien/mal
tambien tenia pensado ponerle poner try/catch al oprimir botones, cosa de que no esten (falta comletar un campo, etc)

didierrenardub replied:
Es buena idea y deberías hacerlo, pero adicionalmente debería probarse que la validación sea correcta. Si te falta completar un campo existente, no debería crashear pero sí debería estar deshabilitado el botón "enviar".

Perfecto lo de los tests atómicos, lo podés hacer en otro archivo.

franciscoMigliardi24 replied:
perfeco, hacer una funcion para completar los imputs, con dentro un try/catch, y luego al momento de apretar el boton, otro try/catch en caso de que un campo este vacio.
gracias

didierrenardub replied:
Lo último no tanto; si el campo está vacío, el try-catch no creo que lo detecte...

franciscoMigliardi24 replied:
por ejemplo:
def completarCamposFunc(idCampo, contenidoCampo):
>try:
	element = driver.find_element_by_name(idCampo)
	element.send_keys(contenidoCampo)
	print("el campo " + idCampo + "fue comletado con " + contenidoCampo)
except:
	print("el campo " + idCampo + "no fue comletado con " + contenidoCampo)

y en el boton 
>try:
	driver.find_element_by_class_name('boton-enviar').click()
	print("se creo el cliente exitosamente")
except:
	print("el cliente no pudo ser creado")

supongamos que se puede tabular aca y no sale todo sobre el mismo margen

franciscoMigliardi24 replied:
el try catch es para buscar el elemento y que no explote en el intento, y si solo fue un campo eliminado que prosiga
por ejemplo, si el correo no es mas util y se elemina el campo, una edicion menos en el .py

didierrenardub replied:
Claro, pero `.click()` no creo que tire una excepción en caso de que el botón esté deshabilitado, a eso me refiero...

franciscoMigliardi24 replied:
ahh, la verdad no investigue... ahora salgo del laburo y me pongo bien con esto
pero entiendo lo que decis, hace click al vacio sino
lo malo es que no tengo un url distinto (dentro de la sesion es solo /logged), como para probar si se refresco la pagina.
ahora veo que puedo inventar

didierrenardub reviewed (changes requested):
Te puse comentarios más que nada en uno de los scripts pero que son aplicables a todos ellos

franciscoMigliardi24 committed:
arreglos pr

didierrenardub commented on crearCliente.py:
```diff
@@ -28,52 +28,52 @@
\ No newline at end of file
+ale = driver.switch_to_alert()
+ale.text
+ale.accept
```
Qué hacen estas dos líneas?

franciscoMigliardi24 replied:
al crear un cliente/modificar/etc etc aparece un alert al cual hay que darle confirmar
este ya tiene el mensaje "usuario creado con exito" o "no se pudo crear usuario" y asi una respuesta por si y por no por cada alta y modificacion
primero cambio el foco a el alert
muestro el mensaje en consola
le doy accept al alert para que salga

didierrenardub replied:
Y no hace falta llamar con paréntesis? Tipo `ale.accept()`?

franciscoMigliardi24 replied:
haciendo mas test, el problema esta en el webdriver.Chrome() y la version de chrome,
se volvio a la alternativa de try catch con firefox que funciona correctamente
proximamente se verndran los cambios correspondientes

franciscoMigliardi24 committed:
arreglos funcion alert((())))

franciscoMigliardi24 committed:
arreglos de pr anteriores

franciscoMigliardi24 committed:
arreglos de webdriver.Chrome() y modificaciones para que funcione correctamente

didierrenardub commented on README.md:
```diff
@@ -1,3 +1,3 @@
 "pip install selenium"
-apretar F5 para ejecutar cada test
\ No newline at end of file
+abrir el archivo en visual code, y luego apretar F5 para ejecutar el mismo
```
Qué archivo hay que abrir en Visual Code? 🙂

franciscoMigliardi24 replied:
jajaj veo que te gusta verme sufrir, lo venis haciendo bien
ahora lo modifico

didierrenardub commented on crearCliente.py:
```diff
@@ -2,31 +2,37 @@
-driver = webdriver.Chrome()
-driver.get("http://localhost:8081/login")
+driver = webdriver.Firefox()
+driver.get("http://localhost:8080/login")
```
Recién me doy cuenta, pero la url base (en este caso, `http://localhost:8080/`) debería poder establecerse en algún archivo de configuración, por si la persona ejecutando las pruebas tiene configurado el entorno de otra forma.

franciscoMigliardi24 replied:
me encantaria decirte que se hacerlo, ahora me pongo a fijar que onda

franciscoMigliardi24 replied:
bueno recien hice unas pruebas y el "driver = webdriver.Firefox()" imposible que este en otro archivo, sino no se inicializa la variable "driver"
pero ya encontre como hacer un archivo config y ahi ingresar el url

didierrenardub commented on crearCliente.py:
```diff
@@ -2,31 +2,37 @@
 
 #iniciar sesion
-elem = driver.find_element_by_name("nombreUsuario")
+elem = driver.find_element_by_id("nombreUsuario")
```
No debería haber sido necesario cambiar estas cosas por el cambio de web driver. Si así es, tenés que dejarlo de tal manera que no importa qué web driver se use, no sea necesario modificar más código.

franciscoMigliardi24 replied:
esto lo hago dependiendo de como este en el front end de vue.js, si el elemento no tiene id o tiene un name repetido se genera conflicto

didierrenardub commented on crearCliente.py:
```diff
@@ -2,31 +2,37 @@
 
-driver = webdriver.Chrome()
-driver.get("http://localhost:8081/login")
+driver = webdriver.Firefox()
```
Ahora que lo pienso, como dije respecto de la url base, esto (el web driver) también debería poder configurarse desde un archivo que no sea de código.

franciscoMigliardi24 replied:
igual que arriba

didierrenardub commented:
No me deja comentar en el archivo, pero así como subiste el geckodriver habría que subir el chromedriver.

franciscoMigliardi24 committed:
arreglos de redme y archivo config

franciscoMigliardi24 committed:
primero, perdon no sabia que estaba en esta branch, me olvide de eso.

detalles del commit:
- se arreglo para que todos los test consuman de test_front.cfg
- se agrego crearFactura.py, crea una factura y la imprime para posteriormente enviarla mediante correro

proximamente (seguro en esta branch) haga el test de eliminar factura y marcar una misma como pagada.
y con eso creo que ya estarian todos los casos, si me falta alguno lo commento en elproximo commit

salu2

franciscoMigliardi24 committed:
arreglo para que al editar el cliente valide el dato previo y no repita

franciscoMigliardi24 committed:
primeras pruebas para eliminar una factura

franciscoMigliardi24 committed:
primeras pruebas para pagar factua

ambos sabemos que es 99% definitivo
pero funciona
